serialization/microsoft_cfb.ksy

```diff
@@ -36,107 +68,173 @@ types:
+      - id: difat_sectors_number
+        type: u4
         doc: Number of DIFAT sectors in this file.
       - id: difat
         repeat: expr
         repeat-expr: 109
-        type: s4
-  fat_entries:
+        type: difat_entry_type
+      - id: sector_padding
+        repeat: expr
+        repeat-expr: _root.sector_size - 0x200
+        type: u4
+        doc: Header is 512 bytes long and if the major version is 4 then remaining bytes in this sector must be filled with zeros.
+    enums:
+      version_type:
+        3: three
+        4: four
+    types:
+      difat_entry_type:
+        seq:
+          - id: sector
+            type: u4
+            enum: sector_type
+        instances:
+          fat:
+            io: _root._io
+            pos: (sector.to_i + 1) * _root.sector_size
+            size: _root.sector_size
+            type: fat_entries_type
+            if: sector != sector_type::free
+      difat_chain_entry_type:
+        seq:
+          - id: sector
+            type: u4
+            enum: sector_type
+        instances:
+          difat:
+            io: _root._io
+            pos: (sector.to_i + 1) * _root.sector_size
+            size: _root.sector_size
+            type: difat_entries_type
+            if: sector != sector_type::endofchain
+      difat_entries_type:
+        seq:
+          - id: entries
+            repeat: expr
+            repeat-expr: _root.sector_size / 4 - 1  # 4 - sizeof<difat_entry_type>
+            type: difat_entry_type
+          - id: chain_entry
+            type: difat_chain_entry_type
+      fat_entries_type:
+        seq:
+          - id: sectors
+            type: u4
+            enum: sector_type
+            repeat: eos
+  mini_fat_entries_type:
     seq:
-      - id: entries
-        type: s4
+      - id: sectors
+        type: u4
+        enum: sector_type
         repeat: eos
-  dir_entry:
+  dir_entries_type:
+    seq:
+      - id: dirs
+        type: dir_entry_type
+        repeat: eos
+  dir_entry_type:
     seq:
       - id: name
         type: str
         size: 64
         encoding: UTF-16LE
-      - id: name_len
+      - id: name_length
         type: u2
       - id: object_type
         type: u1
-        enum: obj_type
+        enum: object_type
       - id: color_flag
         type: u1
         enum: rb_color
       - id: left_sibling_id
-        type: s4
+        type: u4
+        enum: stream_id_type
       - id: right_sibling_id
-        type: s4
+        type: u4
+        enum: stream_id_type
       - id: child_id
-        type: s4
+        type: u4
+        enum: stream_id_type
       - id: clsid
         size: 16
       - id: state
         type: u4
         doc: User-defined flags for storage or root storage objects
-      - id: time_create
+      - id: ctime
         type: u8
         doc: Creation time, in Windows FILETIME format (number of 100-nanosecond intervals since January 1, 1601, UTC)
-      - id: time_mod
+      - id: mtime
```

Comment:
:-1:

Reply:
`time_mod` is better than `mtime`? :) `mtime` and `ctime` are commonly used names for these things (like described [here](https://www.unixtutorial.org/atime-ctime-mtime-in-unix-filesystems/)). Basically I don't mind changing it back. I'm just wondering.

Reply:
probably I better rename these two fields to `modification_time` and `creation_time`

Reply:
Sorry for butting in, but I feel this is a good time to link you to Kaitai's style guide, specifically the section about `id`: https://doc.kaitai.io/ksy_style_guide.html#attr-id

Reply:
well, I read the style guide. and now I have a question, should I revert only time_mod and time_create? or it applies to all my changes?